src/shared/errors.ts

```diff
@@ -0,0 +1,27 @@
+import { responseMsgs } from '../constants';
+
+const buildLatticeResponseErrorMessage = ({ responseCode, errorMessage }) => {
+  const msg: string[] = [];
+  if (responseCode) {
+    msg.push(`Response Code: ${responseCode}`);
+    msg.push(`Message: ${responseMsgs[responseCode]}`);
+  }
+  if (errorMessage) {
+    msg.push('Error Message: ');
+    msg.push(errorMessage);
+  }
+  return msg.join('\n');
+};
+
+export class LatticeResponseError extends Error {
+  constructor(
+    public responseCode: number,
+    public errorMessage: string,
+  ) {
+    const message = buildLatticeResponseErrorMessage({ responseCode, errorMessage });
+    super(message);
+    this.name = 'LatticeResponseError';
```

Comment:
Does this display differently than before? A bit lost in the code at this point, but would you mind posting what a fully bubbled error message would look like to a requester e.g. MM?

Reply:
I didn't consider the implications of changing the error message that could potentially be displayed to the user in the MM context. I'll need to re-evaluate this because the error message _is_ different. 

I added a check for request declined, just to trigger the error. So this isn't a normal error you'd see regularly, but it is an example:

```
LatticeResponseError: Response Code: 132
Message: Request declined by user
Error Message: 
Request declined by user (Lattice)
    at /Users/douglance/Developer/gridplus-sdk/src/shared/functions.ts:125:13
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async sign (/Users/douglance/Developer/gridplus-sdk/src/functions/sign.ts:46:29)
    at async /Users/douglance/Developer/gridplus-sdk/src/__test__/e2e/general.test.ts:100:7
    ...
```

We send a message from the backend, and the SDK itself has its own messages that correspond to each response code. That's why it seems like the message has been duplicated there. Not sure how we want to handle that. 

The reasoning behind this system is that because `LatticeResponseError` is a custom Error handler, it lets us `throw` from anywhere in a lower function, attach some data, then catch it in a higher function. That's how the `retryHandler` is able to catch and use the data we attach here without needing to be tightly coupled to the `request()` function.

Reply:
The `132` and message are redundant. Maybe just:

> LatticeResponseError: Request declined by user

Reply:
991ebaa